app/ts/simulation/services/EthereumClientService.ts

```diff
@@ -212,7 +212,25 @@ export class EthereumClientService {
 		const parentBlock = await this.getBlock(requestAbortController, blockNumber)
 		if (parentBlock === null) throw new Error(`The block ${ blockNumber } is null`)
 
-		const getBlockStateCall = async (block: SimulationStateInputMinimalDataBlock, blockdelta: number) => {
+		const blockOverrides: BlockOverrides[] = []
+		let previousBlockOverride = {
+			time: parentBlock.timestamp,
+			gasLimit: parentBlock.gasLimit,
+			feeRecipient: parentBlock.miner,
+			baseFeePerGas: parentBlock.baseFeePerGas === undefined ? 15000000n : parentBlock.baseFeePerGas
+		}
+		for (const inputBlock of simulationStateInput.blocks) {
+			const getNewBlockTimeStampOverride = (previousBlockTimeStamp: Date, blockTimeManipulation: BlockTimeManipulation) => {
+				const prevTime = BigInt(previousBlockTimeStamp.getTime()) / 1000n
+				if (blockTimeManipulation.type === 'AddToTimestamp') return new Date(Number(prevTime + blockTimeManipulation.deltaToAdd) * 1000)
+				return new Date(Number(max(prevTime, blockTimeManipulation.timeToSet)) * 1000)
```

Comment:
I wonder if we should add a utility method for `bigintSecondsToDate` and `dateToBigintSeconds`?

Reply:
Looks like there are other places where this would be beneficial.

Reply:
yeah I think that's good idea